OpenSim/Auxiliary/auxiliaryTestFunctions.h

```diff
@@ -43,6 +43,9 @@ void ASSERT_EQUAL(T expected,
                   std::string message = "") {
     if (found < expected - tolerance || found > expected + tolerance)
         throw OpenSim::Exception(message, file, line);
+    // A variable is not equivalent to itself only if it is NaN
+    else if( (found != found)  && (expected != expected) )
```

Comment:
Three questions here:
1. Why is this `&&` rather than `||` (i.e., throw if *either* `found` *or* `expected` is NaN)? I don't think `ASSERT_EQUAL` should report that `5` is equal to `NaN`.
2. Why not use `SimTK::isNaN()` rather than the somewhat less readable `if (a != a)`?
3. I would have expected this condition to be caught before doing anything else:
```cpp
{
    if ( SimTK::isNaN(found) || SimTK::isNaN(expected) )
        throw ...
    // inputs are valid; proceed with algorithm
    ...
}
```